ni_measurementlink_service/measurement/service.py

```diff
@@ -370,12 +375,20 @@ def _make_annotations_dict(
 
         return annotations
 
-    def _enum_to_annotations_value(self, enum_type: Type[Enum]) -> str:
-        if not any(member.value == 0 for member in enum_type):
-            raise ValueError("The enum does not have a value for 0.")
+    def _enum_to_annotations_value(
+        self, enum_type: Union[Type[Enum], Type[EnumTypeWrapper]]
+    ) -> str:
         enum_values = {}
-        for member in enum_type:
-            enum_values[member.name] = member.value
+        if isinstance(enum_type, EnumTypeWrapper):
```

Comment:
Should this be `issubclass`? Or is an instance of `EnumTypeWrapper` itself a type?

Reply:
The enum_type that gets passed in for protobuf enums appears to be an instance of EnumTypeWrapper in addition to being a type itself which is why we used `isinstance` here. I'm not entirely sure why the protobuf enum types are instances of EnumTypeWrapper rather than just being a subtype of it, but that seems to be the way it behaves. 

In our decorator, the intent was for users to pass in that type directly.

`@sample_measurement_service.configuration("Protobuf Enum In", nims.DataType.Enum, ProtobufColor.BLACK, enum_type=ProtobufColor)`

If instead we wanted to use `issubclass(enum_type, EnumTypeWrapper)` here, we would need the users to pass in `enum_type=type(ProtobufColor))` instead. 

I updated this code to check `issubclass(type(enum_type), EnumTypeWrapper)` and added a comment to make it clearer, but note that `type(enum_type)` is required since `issubclass` throws an error if the parameter is an instance.

Reply:
I'm not sure whether the enum type wrapper instance is a type or not.

The `.pyi` file says it subclasses `builtins.type`:
https://github.com/ni/nidaqmx-python/blob/master/generated/nidaqmx/_stubs/session_pb2.pyi

But at run time, `isinstance` says it's not a type.
```
>>> from nidaqmx._stubs import nidaqmx_pb2
>>> session_pb2.SessionInitializationBehavior
<google.protobuf.internal.enum_type_wrapper.EnumTypeWrapper object at 0x000001D87531E8E0>
>>> session_pb2.SessionInitializationBehavior.keys()
['SESSION_INITIALIZATION_BEHAVIOR_UNSPECIFIED', 'SESSION_INITIALIZATION_BEHAVIOR_INITIALIZE_NEW', 'SESSION_INITIALIZATION_BEHAVIOR_ATTACH_TO_EXISTING']
>>> type(session_pb2.SessionInitializationBehavior)
<class 'google.protobuf.internal.enum_type_wrapper.EnumTypeWrapper'>
>>> isinstance(session_pb2.SessionInitializationBehavior, type)
False
```

I wonder if `SupportedEnumType = Union[Type[Enum], EnumTypeWrapper]` would also work.

Reply:
Yeah, I saw this too and I'm not entirely sure why it behaves this way. Despite the `type()` of it being a class, issubclass does not believe it to be so. 

![image](https://github.com/ni/measurementlink-python/assets/103216564/a27527f1-b815-4b1c-ac10-18c470bb16e7)

Note that `SupportedEnumType = Union[Type[Enum], EnumTypeWrapper]` does also work. Let me know if you prefer this to `Union[Type[Enum], Type[EnumTypeWrapper]]` and I can make the change, but I've left it as-is for now since I'm not sure it really makes it any clearer.

Reply:
Yes, I prefer treating it as an instance of `EnumTypeWrapper` because that is true for both run time and type checking.

Also, please remove "in addition to being a type itself" from the comment because this is only true for type checking, not run time.

If you needed to change the order of the if-statement cases, I think `if isinstance(enum_type, type) and issubclass(enum_type, Enum)` would prevent it from throwing.

Reply:
Switched to `Union[Type[Enum], EnumTypeWrapper]` as suggested and fixed comment. Fixed condition as suggested to fix mypy error as suggested.